gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HttpHook.java

```diff
@@ -302,6 +305,27 @@ public void setMaxWaitQueueSize(Integer maxWaitQueueSize) {
         this.maxWaitQueueSize = maxWaitQueueSize;
     }
 
+    /**
+     * @return Timeout for request in milliseconds. Defines the time up until
+     *         an ongoing request will be aborted if there was no reply given before.
+     *         This may returning null in case there's no value
+     *         specified. Callers may catch that by fall back to a default.
+     */
+    @Nullable
+    public Integer getTimeout() {
+        return this.timeout;
+    }
+
+    /**
+     * See {@link #getTimeout()}.
+     */
+    public void setTimeout(Integer timeout) {
+        if (timeout != null && timeout < -1){
+            throw new IllegalArgumentException("Values below -1 not valid.");
```

Comment:
Is a timeout value of **-1** allowed? Should it not always be a positive number?

Reply:
Right, will improve this.